src/main/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -283,7 +288,8 @@ public int read(final byte[] buffer, int offset, int length)
     public void seek(final long pos)
         throws IOException {
         if (mFile == null) {
-            throw new IOException("Cannot seek on stream based file");
+
+            throw new IOException("Cannot seek on stream based file " + mFile.getSource());
```

Comment:
Done. Cannot get the mFile path since mFile is null. The path is not accessible form mStream.